src/urls.ts

```diff
@@ -22,27 +22,49 @@ export const getWorkUrl = ({
   return workUrl;
 };
 
-export const getAsShortUrl = ({ url }: { url: string }) => url.replace(/archiveofourown/, 'ao3');
+export const getAsShortUrl = ({ url }: { url: string }) =>
+  url.replace(/archiveofourown/, "ao3");
 
 export const getUserProfileUrl = ({ username }: { username: string }) =>
   `https://archiveofourown.org/users/${encodeURI(username)}/profile`;
 
-const tokenReplacerObject = {
-  '/': '*s*',
-  '&': '*a*',
-  '.': '*d*',
-  '#': '*h*',
-  '?': '*q*'
-}
-const tokensToEscape = ['/', '?', '.']
-const replaceEscapableTokens = (c:string) => tokensToEscape.includes(c) ? `\\${c}` : c;
-const tagUrlReplaceChars = new RegExp(`(${Object.keys(tokenReplacerObject).map(replaceEscapableTokens).join('|')})`, 'g')///(\/|\.|&|#|\?)/g;
+const TOKEN_REPLACEMENTS_MAP = {
+  "/": "*s*",
+  "&": "*a*",
+  ".": "*d*",
+  "#": "*h*",
+  "?": "*q*",
+} as const;
 
-type ReplacerObjectKeys = keyof typeof tokenReplacerObject
+type ReplaceableToken = keyof typeof TOKEN_REPLACEMENTS_MAP;
```

Comment:
Similarly, a "key" here is a token that can be replaced. Whether you hand-typed an array with them or got them from the key of an object doesn't matter, really, so I prefer a direct statement.